### PR TITLE
[WIP] first sketch of a garbage collector

### PR DESCRIPTION
**Pull request by noha**

### CODE REVIEW

File: src/Soil-Core/SoilGarbageCollectVisitor.class.st [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{ #category : #visiting }
+SoilGarbageCollectVisitor >> visitSkipList: aSoilSkipList [ 
+	| newSkipList |
+	indexIds add: aSoilSkipList indexId.
+	newSkipList := newSegment indexManager  
+		createIndexWithId: aSoilSkipList indexId 
+		class: aSoilSkipList class. 
+	newSkipList initializeParametersFrom: aSoilSkipList. 
+	
+	aSoilSkipList newIterator basicAssociationsDo: [ :item |
```
Comment (MarcusDenker) [inline src/Soil-Core/SoilGarbageCollectVisitor.class.st:89]: #basicAssociationsDo: was renamed to #associationsDo:

File: src/Soil-Core/SoilGarbageCollectVisitor.class.st [hunk 2]
```diff
@@ -0,0 +1,94 @@
+Class {
+	#name : #SoilGarbageCollectVisitor,
+	#superclass : #SoilCloneVisitor,
+	#instVars : [
+		'newRepository',
```
Comment (MarcusDenker) [inline src/Soil-Core/SoilGarbageCollectVisitor.class.st:5]: newRepository is not used (clean code test fails due to that)

**Comment by noha** [comment]
this is WIP and I just resurrected the branch and wanted to keep it. This is not the time a review is needed. thanks!

**Comment by noha** [comment]
But you can just commit the fixes, I don't mind

**Comment by noha** [comment]
Clsing this as it will be committed along with #882

**Pull request closed without merging**